File: mergeinterval/meetingrooms/AttendMeetingsChk.java
package meetingrooms;

import java.util.Arrays;
class AttendMeetingsChk {
    public boolean canAttendMeetings(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));

        for (int i = 1; i < intervals.length; i++) {
            if (intervals[i][0] < intervals[i-1][1]) {
                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) {
        AttendMeetingsChk solution = new AttendMeetingsChk();
        System.out.println(solution.canAttendMeetings(new int[][]{{0,30},{5,10},{15,20}}));  // false
        System.out.println(solution.canAttendMeetings(new int[][]{{7,10},{2,4}}));  // sorted to {2,4},{7,10} : true
    }
}

File: mergeinterval/meetingrooms/AttendMeetingsChk.py
class AttendMeetingsChk:
    def can_attend_meetings(self, intervals):
        # Sort intervals by start time
        intervals.sort(key=lambda x: x[0])

        # Check for overlaps
        for i in range(1, len(intervals)):
            if intervals[i][0] < intervals[i-1][1]:
                return False

        return True

# Testing the code
if __name__ == "__main__":
    solution = AttendMeetingsChk()
    print(solution.can_attend_meetings([[0, 30], [5, 10], [15, 20]]))  # Output: False
    print(solution.can_attend_meetings([[7, 10], [2, 4]]))  # Output: True
